### ISSUE TITLE
Helmsman 1.0.1 failure with MAF file

### ISSUE THREAD

**Issue by ashipde** [carjed/helmsman#6]
I am unable to use Helmsman on a MAF file from GDC for TCGA data (http://api.gdc.cancer.gov/data/1c8cfe5f-e52d-41ba-94da-f15ea1337efc). The MAF uses 'Start_Position' and 'SNV', and this causes problem because Helmsman 1.0.1 has in its code (util.py):

```
line 361: if(row['Variant_Type'] != "SNP")
line 363: pos = int(row['Start_position'])
```

Even after fixing the code, Helmsman quits with:

```
[helmsman::<module>] INFO ----------------------------------
[helmsman::<module>] INFO helmsman.py 1.0.1
[helmsman::<module>] INFO ----------------------------------
[helmsman::<module>] INFO random seed: 949579
Traceback (most recent call last):
  File "helmsman.py", line 355, in <module>
    writeM(M, paths['M_path'], subtypes_dict, samples)
  File "/home/spatnaik/software/helmsman/util.py", line 619, in writeM
    columns=list(sorted(subtypes_dict.keys())))
  File "/home/spatnaik/miniconda2/envs/helmsman/lib/python3.6/site-packages/pandas/core/frame.py", line 379, in __init__
    copy=copy)
  File "/home/spatnaik/miniconda2/envs/helmsman/lib/python3.6/site-packages/pandas/core/frame.py", line 536, in _init_ndarray
    return create_block_manager_from_blocks([values], [columns, index])
  File "/home/spatnaik/miniconda2/envs/helmsman/lib/python3.6/site-packages/pandas/core/internals.py", line 4859, in create_block_manager_from_blocks
    mgr = BlockManager(blocks, axes)
  File "/home/spatnaik/miniconda2/envs/helmsman/lib/python3.6/site-packages/pandas/core/internals.py", line 3282, in __init__
    self._verify_integrity()
  File "/home/spatnaik/miniconda2/envs/helmsman/lib/python3.6/site-packages/pandas/core/internals.py", line 3498, in _verify_integrity
    len(self.items), tot_items))
AssertionError: Number of manager items must equal union of block items
# manager items: 96, # tot_items: 0
```

**Comment by carjed** [comment]
The TCGA file provided (mc3.v0.2.8.PUBLIC.maf) does appear to use "SNP" and not "SNV" in the Variant_Type field, but I'll add a fix in the next release to check for either string when processing MAF files.

After fixing the case issue with the Start_position field in line 363, I could process this MAF file successfully and was unable reproduce the error. Can you post the changes you made to the code and the output when running with the `--verbose` flag?

**Comment by ashipde** [comment]
My environment is a 64-bit Linux on an HPCC. I used Miniconda (for Python 3.7) for installing Helmsman.

```
git clone https://github.com/carjed/helmsman.git
# The code changes I made to util.py are on the two lines noted in my first post.
conda env create -n helmsman -f env.yml
source activate helmsman
```

My Helmsman command:

`python helmsman.py --verbose --cpus 8 --mode maf --input mc3.v0.2.8.PUBLIC.maf --fastafile GRCh37.p13_genomic.fa --projectdir out --verbose`

Output:

```
/home/spatnaik/miniconda3/envs/helmsman/lib/python3.6/site-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
[helmsman::<module>] INFO ----------------------------------
[helmsman::<module>] INFO helmsman.py 1.0.1
[helmsman::<module>] INFO ----------------------------------
[helmsman::<module>] DEBUG Running with the following options:
[helmsman::<module>] DEBUG cpus: 8
[helmsman::<module>] DEBUG seed: 1042418
[helmsman::<module>] DEBUG verbose: True
[helmsman::<module>] DEBUG mode: maf
[helmsman::<module>] DEBUG input: mc3.v0.2.8.PUBLIC.maf
[helmsman::<module>] DEBUG rowwise: False
[helmsman::<module>] DEBUG fastafile: GRCh37.p13_genomic.fa
[helmsman::<module>] DEBUG samplefile: None
[helmsman::<module>] DEBUG groupvar: Tumor_Sample_Barcode
[helmsman::<module>] DEBUG minsnvs: 0
[helmsman::<module>] DEBUG maxac: 0
[helmsman::<module>] DEBUG projectdir: y
[helmsman::<module>] DEBUG matrixname: subtype_count_matrix
[helmsman::<module>] DEBUG package: None
[helmsman::<module>] DEBUG decomp: None
[helmsman::<module>] DEBUG rank: 0
[helmsman::<module>] DEBUG length: 3
[helmsman::<module>] INFO random seed: 1042418
[util::indexSubtypes] DEBUG 96 3-mer subtypes indexed
Traceback (most recent call last):
  File "helmsman.py", line 355, in <module>
    writeM(M, paths['M_path'], subtypes_dict, samples)
  File "/home/spatnaik/software/helmsman/util.py", line 619, in writeM
    columns=list(sorted(subtypes_dict.keys())))
  File "/home/spatnaik/miniconda3/envs/helmsman/lib/python3.6/site-packages/pandas/core/frame.py", line 379, in __init__
    copy=copy)
  File "/home/spatnaik/miniconda3/envs/helmsman/lib/python3.6/site-packages/pandas/core/frame.py", line 536, in _init_ndarray
    return create_block_manager_from_blocks([values], [columns, index])
  File "/home/spatnaik/miniconda3/envs/helmsman/lib/python3.6/site-packages/pandas/core/internals.py", line 4859, in create_block_manager_from_blocks
    mgr = BlockManager(blocks, axes)
  File "/home/spatnaik/miniconda3/envs/helmsman/lib/python3.6/site-packages/pandas/core/internals.py", line 3282, in __init__
    self._verify_integrity()
  File "/home/spatnaik/miniconda3/envs/helmsman/lib/python3.6/site-packages/pandas/core/internals.py", line 3498, in _verify_integrity
    len(self.items), tot_items))
AssertionError: Number of manager items must equal union of block items
# manager items: 96, # tot_items: 0
```

**Comment by carjed** [comment]
Thanks for your help with debugging this--can you post the output of `conda list` after activating the helmsman environment? I suspect this may be an issue with the specific libraries (or versions thereof) getting loaded in the virtual environment in Miniconda.

**Comment by ashipde** [comment]
Thanks for looking into this. Conda list:
```

# packages in environment
# Name                    Version                   Build  Channel
bcftools                  1.6                           1    bioconda
binutils_impl_linux-64    2.31.1               h6176602_1  
binutils_linux-64         2.31.1               h6176602_3  
biopython                 1.72             py36h470a237_0    conda-forge
blas                      1.1                    openblas    conda-forge
bzip2                     1.0.6                h470a237_2    conda-forge
ca-certificates           2018.10.15           ha4d7672_0    conda-forge
certifi                   2018.10.15            py36_1000    conda-forge
click                     7.0                        py_0    conda-forge
colorama                  0.4.0                      py_0    conda-forge
coloredlogs               10.0                  py36_1000    conda-forge
curl                      7.62.0               h74213dd_0    conda-forge
cycler                    0.10.0                     py_1    conda-forge
cyvcf2                    0.8.4            py36h355e19c_4    bioconda
dbus                      1.13.0               h3a4f0e9_0    conda-forge
expat                     2.2.5                hfc679d8_2    conda-forge
fontconfig                2.13.1               h65d0f4c_0    conda-forge
freetype                  2.9.1                h6debe1e_4    conda-forge
gcc_impl_linux-64         7.3.0                habb00fd_1  
gcc_linux-64              7.3.0                h553295d_3  
gettext                   0.19.8.1             h5e8e0c9_1    conda-forge
gfortran_impl_linux-64    7.3.0                hdf63c60_1  
gfortran_linux-64         7.3.0                h553295d_3  
glib                      2.56.2               h464dc38_1    conda-forge
gst-plugins-base          1.12.5               hde13a9d_0    conda-forge
gstreamer                 1.12.5               h5856ed1_0    conda-forge
gxx_impl_linux-64         7.3.0                hdf63c60_1  
gxx_linux-64              7.3.0                h553295d_3  
humanfriendly             4.16.1                     py_0    conda-forge
icu                       58.2                 hfc679d8_0    conda-forge
joblib                    0.11                     py36_0    conda-forge
jpeg                      9c                   h470a237_1    conda-forge
kiwisolver                1.0.1            py36h2d50403_2    conda-forge
krb5                      1.16.2               hbb41f41_0    conda-forge
libcurl                   7.62.0               hbdb9355_0    conda-forge
libedit                   3.1.20170329         haf1bffa_1    conda-forge
libffi                    3.2.1                hfc679d8_5    conda-forge
libgcc                    7.2.0                h69d50b8_2    conda-forge
libgcc-ng                 8.2.0                hdf63c60_1  
libgfortran               3.0.0                         1    conda-forge
libgfortran-ng            7.3.0                hdf63c60_0  
libiconv                  1.15                 h470a237_3    conda-forge
libpng                    1.6.35               ha92aebf_2    conda-forge
libssh2                   1.8.0                h5b517e9_2    conda-forge
libstdcxx-ng              8.2.0                hdf63c60_1  
libuuid                   2.32.1               h470a237_2    conda-forge
libxcb                    1.13                 h470a237_2    conda-forge
libxml2                   2.9.8                h422b904_5    conda-forge
matplotlib                3.0.1                h8a2030e_1    conda-forge
matplotlib-base           3.0.1            py36hc039c98_1    conda-forge
ncurses                   6.1                  hfc679d8_1    conda-forge
nimfa                     1.3.2                     <pip>
numpy                     1.15.4          py36_blas_openblashb06ca3d_0  [blas_openblas]  conda-forge
openblas                  0.3.3                ha44fe06_1    conda-forge
openssl                   1.0.2p               h470a237_1    conda-forge
pandas                    0.23.4           py36hf8a1672_0    conda-forge
pcre                      8.41                 hfc679d8_3    conda-forge
pip                       18.1                  py36_1000    conda-forge
pthread-stubs             0.4                  h470a237_1    conda-forge
pyfaidx                   0.5.5.2                    py_0    bioconda
pyparsing                 2.3.0                      py_0    conda-forge
pyqt                      5.6.0            py36h8210e8a_7    conda-forge
python                    3.6.6                h5001a0f_3    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
pytz                      2018.7                     py_0    conda-forge
qt                        5.6.2                hf70d934_9    conda-forge
readline                  7.0                  haf1bffa_1    conda-forge
scikit-learn              0.20.0          py36_blas_openblash00c3548_201  [blas_openblas]  conda-forge
scipy                     1.1.0           py36_blas_openblashb06ca3d_202  [blas_openblas]  conda-forge
setuptools                40.5.0                   py36_0    conda-forge
sip                       4.18.1           py36hfc679d8_0    conda-forge
six                       1.11.0                py36_1001    conda-forge
sqlite                    3.25.3               hb1c47c0_0    conda-forge
tk                        8.6.8                ha92aebf_0    conda-forge
tornado                   5.1.1            py36h470a237_0    conda-forge
wheel                     0.32.2                   py36_0    conda-forge
xorg-libxau               1.0.8                h470a237_6    conda-forge
xorg-libxdmcp             1.1.2                h470a237_7    conda-forge
xz                        5.2.4                h470a237_1    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge
```

**Comment by carjed** [comment]
I was able to reproduce these errors after changing `if(row['Variant_Type'] != "SNP")` to `if(row['Variant_Type'] != "SNV")`. As mentioned in my initial comment, the variants in the test file are all coded as "SNP" and not "SNV," so this change was unnecessary and actually caused all variants in the MAF file to be skipped, leaving pandas with an empty data frame, hence the errors.

When fixing this, I realized that I had not included any way to handle inconsistencies in how the fasta file and input file format the chromosome fields. In this case, the fasta file prepended each chromosome number with "chr" but the MAF file did not, and caused an entirely different error that you probably encountered at some point (something like `KeyError: '10 not in GRCh37.p13.genome.fa.'`).

The latest release (1.1.0) should resolve the chromosome formatting mismatch issue. I've also added the ability to check for either "SNV" or "SNP" in the `Variant_Type` field, as well as checking the case of the `Start_position` field. 

Let me know if you're still running into problems!
